### ISSUE TITLE
Server Disconnect on leaving rideable entity

### ISSUE THREAD

**Issue by Y0ungSandwich** [SpongePowered/SpongeForge#2301]
**I am currently running**

- SpongeForge version: 1.12.2-2705-7.1.0-BETA-3206
- Forge version: 1.12.2-2705
- Java version: 8
- Operating System:CentOS

- Plugins/Mods:Nucleus, Villager Shops, Grief Prevention, EconomyLite, LuckPerms

**Issue Description**
Any rideable entity, i.e. Boats, ViesCraft airships, horses, etc, will disconnect the player from the server upon getting off. Player shooting skeleton riders off other mobs will also be disconnected from server.
When reconnecting the player will remain on whatever they were riding on before the disconnect. 

https://gist.github.com/Y0ungSandwich/fad165fee875a00e95e5225b754ed3c9

**Comment by Aaron1011** [comment]
This is *yet another* mod that's directly modifying the world from its Netty handler:

```
java.lang.IllegalStateException: CauseStackManager called from off main thread (current='Thread{class=class java.lang.Thread, name=Netty Epoll Server IO #8, priority=5, group=net.minecraftforge.fml.common.thread.SidedThreadGroup[name=SERVER,maxpri=10]}', expected='Thread{class=class java.lang.Thread, name=Server thread, priority=5, group=net.minecraftforge.fml.common.thread.SidedThreadGroup[name=SERVER,maxpri=10]}')!
	at org.spongepowered.common.event.SpongeCauseStackManager.enforceMainThread(SpongeCauseStackManager.java:71) ~[SpongeCauseStackManager.class:1.12.2-2705-7.1.0-BETA-3206]
	at org.spongepowered.common.event.SpongeCauseStackManager.pushCauseFrame(SpongeCauseStackManager.java:141) ~[SpongeCauseStackManager.class:1.12.2-2705-7.1.0-BETA-3206]
	at net.minecraft.entity.Entity.dismountRidingEntity(Entity.java:5163) ~[vg.class:?]
	at net.minecraft.entity.Entity.func_184210_p(Entity.java:5155) ~[vg.class:?]
	at net.minecraft.entity.EntityLivingBase.func_184210_p(EntityLivingBase.java:2470) ~[vp.class:?]
	at net.minecraft.entity.player.EntityPlayer.func_184210_p(EntityPlayer.java:1244) ~[aed.class:?]
	at net.minecraft.entity.player.EntityPlayerMP.func_184210_p(EntityPlayerMP.java:791) ~[oq.class:?]
	at com.zeitheron.hammercore.net.internal.PacketStopRiding2.executeOnServer(PacketStopRiding2.java:39) ~[PacketStopRiding2.class:?]
	at com.zeitheron.hammercore.net.PacketHolder.execute(PacketHolder.java:83) ~[PacketHolder.class:?]
	at com.zeitheron.hammercore.net.HCNet.packetToServer(HCNet.java:234) ~[HCNet.class:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_1076_HCNet_packetToServer_ServerCustomPacketEvent.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) ~[ASMEventHandler.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:662) ~[EventBus.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:597) ~[EventBus.class:?]
	at net.minecraftforge.fml.common.network.FMLEventChannel.fireRead(FMLEventChannel.java:130) ~[FMLEventChannel.class:?]
	at net.minecraftforge.fml.common.network.NetworkEventFiringHandler.channelRead0(NetworkEventFiringHandler.java:51) ~[NetworkEventFiringHandler.class:?]
	at net.minecraftforge.fml.common.network.NetworkEventFiringHandler.channelRead0(NetworkEventFiringHandler.java:39) ~[NetworkEventFiringHandler.class:?]
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105) ~[SimpleChannelInboundHandler.class:?]
```

@Y0ungSandwich For a real fix, you should report this to the mod developer. However, https://github.com/SpongePowered/SpongeForge/pull/2299 will provide a temporary fix once I merge it.

**Comment by Aaron1011** [comment]
I've opened an issue here; https://gitlab.com/DragonForge/HammerCore/issues/69

**Comment by Aaron1011** [comment]
@Y0ungSandwich: The next build of SpongeForge will contain support for working around this issue. in `config/sponge/global.conf`, set `modules.broken-mod=true`. Then, under the `broken-mods` section, add `hammercore` to `broken-network-handler-mods`

**Comment by Aaron1011** [comment]
@Y0ungSandwich: The [issue has now been fixed in HammerCore 2.0.1.1](https://gitlab.com/DragonForge/HammerCore/issues/69#note_89851475). You can disable the `broken-mod` module in your `global.conf` once you upgrade.
